include/robot_localization/ros_filter.h

```diff
@@ -586,6 +586,11 @@ template<class T> class RosFilter
     //! @brief condition variable for the measurement queue
     //!
     boost::condition_variable measurementsReady_;
+
+    //! @brief mutex for diagnostic maps
+    //!
+    boost::mutex diagnosticMapsMutex_;
```

Comment:
Can we add a comment just stating what variables this is protecting. I think this would be a good practice going forward in all code as the number of mutexs in our code is increasing and it can become difficult to figure out exact what they are trying to protect.